### PR TITLE
Vbs elastic pool changes and regenerating sdk for sql

### PR DESCRIPTION
**Pull request by vidit-msft**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by vidit-msft** [comment]
@live1206 Can you please review this PR once. It is currently failing on spell-check for Euap( Prod canary region) I added for recording tests.

image

File: sdk/core/Azure.Core/src/AzureLocation.cs [hunk 1]
```diff
@@ -299,6 +299,16 @@ namespace Azure.Core
         /// </summary>
         public static AzureLocation USGovIowa { get; } = CreateStaticReference("usgoviowa", "US Gov Iowa");
 
+		/// <summary>
```
Comment (live1206) [inline sdk/core/Azure.Core/src/AzureLocation.cs:302]: Why change in `Azure.Core`?

Comment (live1206) [reply sdk/core/Azure.Core/src/AzureLocation.cs:302]: As discussed offline, if you wan to release a public version of .NET SDK, your service should be available publicly.
No need to add Canary region for test here.

Re-generate the code to latest:
```
Generated code is not up to date.
    You may need to rebase on the latest main, 
    run 'eng\scripts\Update-Snippets.ps1' if you modified sample snippets or other *.md files (https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#updating-sample-snippets), 
    run 'eng\scripts\Export-API.ps1' if you changed public APIs (https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#public-api-additions). 
    run 'dotnet build /t:GenerateCode' to update the generated code.
    run 'dotnet build /t:GenerateTests' to update the generated test code.
    
To reproduce this error locally, run 'eng\scripts\CodeChecks.ps1 -ServiceDirectory sqlmanagement'.
```

Compile errors:
```
##[error]sdk/sqlmanagement/Azure.ResourceManager.Sql/src/Generated/ArmSqlModelFactory.cs(2456,24): Error CS7036: There is no argument given that corresponds to the required parameter 'highAvailabilityReplicaCount' of 'ElasticPoolData.ElasticPoolData(ResourceIdentifier, string, ResourceType, SystemData, IDictionary<string, string>, AzureLocation, SqlSku, string, ElasticPoolState?, DateTimeOffset?, long?, double?, ElasticPoolPerDatabaseSettings, bool?, ElasticPoolLicenseType?, ResourceIdentifier, int?, SqlAlwaysEncryptedEnclaveType?, SqlAvailabilityZoneType?)'

/mnt/vss/_work/1/s/sdk/sqlmanagement/Azure.ResourceManager.Sql/src/Generated/ArmSqlModelFactory.cs(2456,24): error CS7036: There is no argument given that corresponds to the required parameter 'highAvailabilityReplicaCount' of 'ElasticPoolData.ElasticPoolData(ResourceIdentifier, string, ResourceType, SystemData, IDictionary<string, string>, AzureLocation, SqlSku, string, ElasticPoolState?, DateTimeOffset?, long?, double?, ElasticPoolPerDatabaseSettings, bool?, ElasticPoolLicenseType?, ResourceIdentifier, int?, SqlAlwaysEncryptedEnclaveType?, SqlAvailabilityZoneType?)' [/mnt/vss/_work/1/s/sdk/sqlmanagement/Azure.ResourceManager.Sql/src/Azure.ResourceManager.Sql.csproj]
```

**Review by live1206 — CHANGES_REQUESTED** [review]
Please resolve the comments for further review.

**Pull request closed without merging**